### PR TITLE
Fix macrobot advertisements

### PR DESCRIPTION
**Pull request by joshblum**
patch does the following:
- force non team advertisements to be to a specific `conv`
- only advertise `create-for-channel` command to teams
-

### CODE REVIEW

**Comment by joshblum** [comment]
@marceloneil gonna merge, happy to address any comments

File: macrobot/macrobot/handler.go [hunk 1]
```diff
@@ -36,6 +36,12 @@ func NewHandler(stats *base.StatsRegistry, kbc *kbchat.API, debugConfig *base.Ch
 }
 
 func (h *Handler) HandleNewConv(conv chat1.ConvSummary) error {
+	// When we're put into a team conv, tell the team about the `create-for-channel` option.
+	if conv.Channel.MembersType == "team" && conv.IsDefaultConv {
```
Comment (marceloneil) [inline macrobot/macrobot/handler.go:40]: Will this trigger if a restricted bot is only added to a specific channel?

File: macrobot/macrobot/handler.go [hunk 2]
```diff
@@ -213,31 +225,39 @@ func (h *Handler) doPrivateAdvertisement(msg chat1.MsgSummary) error {
...
-	} else {
-		if err = h.kbc.ClearCommands(&chat1.ClearCommandAPIParam{
-			Typ:      "teamconvs",
-			TeamName: msg.Channel.Name,
-		}); err != nil {
-			return err
+	if channel.MembersType == "team" {
+		// only teams can see the create-for-channel command
+		teamCmds = append(teamCmds, getCreateForChannelCmd())
+		if len(teamCmds) > 0 {
```
Comment (marceloneil) [inline macrobot/macrobot/handler.go:233]: Won't `len(teamCmds) > 0` always be true?

File: macrobot/macrobot/handler.go (hunk 1 above)
Comment (joshblum) [reply macrobot/macrobot/handler.go:40]: i'll put an lru in for if we've advertised to a conv before instead

**Pull request merged**